dev_tools/scripts/db_integrity_test.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Insert some test data in the db
+docker exec db psql -U postgres -d test -c "INSERT INTO local_ega.main(submission_file_path,submission_user,submission_file_extension,status,encryption_method) VALUES('test.c4gh','dummy','c4gh','INIT','CRYPT4GH');"
+
+# Build, run and execute the pg_basebackup action
+docker container run --rm -i --name pg-backup --network=host $(docker build --build-arg USER_ID=$(id -u) -f dev_tools/Dockerfile-backup -q -t backup .) /bin/sda-backup --action pg_basebackup
+
+# Find the name of the copy in the S3
+DBCOPY=$(s3cmd -c dev_tools/s3conf ls s3://dumps/ | grep ".tar" | cut -d '/' -f4)
+
+# Checks if the backup image exists locally from the previou step, if not creates the image
+# Then gets the physical copy of the database from the S3
+IMAGENAME=$(docker images -q backup)
+NAMELENGTH=`echo -n "$IMAGENAME" | wc -m`
+
+if [ $NAMELENGTH = 0 ]; then
+    docker container run -v $(pwd)/tmp:/home --rm -i --name pg-backup --network=host $(docker build --build-arg USER_ID=$(id -u) -f dev_tools/Dockerfile-backup -q .) /bin/sda-backup --action pg_bb-restore --name "$DBCOPY"
+else
+    docker container run -v $(pwd)/tmp:/home --rm -i --name pg-backup --network=host backup /bin/sda-backup --action pg_bb-restore --name "$DBCOPY"
+fi
+
+# Remove everything from the db
+docker exec -i db /bin/sh -c "rm -r data/pgdata"
+
+# Add the physical copy in the db container
+docker cp tmp/db-backup/ db:data/pgdata
+
+# Check when the container restarts
+docker events --filter 'container=db'  | while read event
+do
+    check_event=$(echo $event | awk '{print $3}')
+    if [ "$check_event" = "start" ]; then
+        echo "container restarted"
+        break
+    fi
+done;
+
```

Comment:
There is no need to delete stuff like this.

Reply:
For the physical copy of the database is not easy to follow the steps you describe above because for sure you need to stop the db container and run it again by pointing to the backup data, which means that an extra compose file would be necessary and I do not know if it is optimal to have another file for this test. Since it is a script for testing I decided to make more rough and just delete the pgdata and copy the backup.
Except the solution with another compose file I thought to try to commit a new image from the db container, then stop the container, delete the db volume manually (since it cannot be done by using the compose file) and run the container from the new committed image and mount a new volume (the backup) to it but there were problems since it depends on the `tlsFixer` and it was failing due to certificates. Probably there is a workaround on that but I decided not to spent more time.
Do you think that I should do one of the two above solutions? Do you know any other way? Or should we just leave it as it is for now?

Reply:
You could mimic a more realistic environment by using docker volumes, then the DB filesystem can be mounted into the container running the restoration

Reply:
I have created an extra compose file with only the `db` and `tlsFixer` services where I mount the backup in the container. It works fine if this is what you mean. The file looks like this:
`version: "3.7"
services:
  db:
    depends_on:
      - tlsFixer
    container_name: db
    command: -c ssl=on -c ssl_cert_file=/data/certs/server.crt -c ssl_key_file=/data/certs/server.key -c ssl_ca_file=/data/certs/ca.pem
    environment:
      - POSTGRES_PASSWORD=postgres
      - PGDATA=/data/pgdata
      - POSTGRES_DB=test
    healthcheck:
      test: ["CMD", "pg_isready", "-h", "localhost", "-U", "postgres"]
      interval: 5s
      timeout: 20s
      retries: 3
    restart: always
    image: postgres:13-alpine
    ports:
      - "5432:5432"
    volumes:
      - ./../tmp/db-backup:/data/pgdata
  tlsFixer:
    image: alpine:3.12
    container_name: fixer
    command: sh -c "mkdir -p /data/certs/ && cp /certs/* /data/certs && chmod 600 /data/certs/*.key && chown 70:70 /data/certs/*"
    volumes:
      - ./certs:/certs:ro`

I had done this since the beginning but I thought that extra files for testing are not desirable